src/Utils/ProductGalleryUtils.php

```diff
@@ -93,4 +102,36 @@ public static function get_product_gallery_image_ids( $product, $max_number_of_v
 
 		return $unique_image_ids;
 	}
+
+	/**
+	 * Generates the intermediate image sizes only when needed.
+	 *
+	 * @param int	 $attachment_id Attachment ID.
+	 * @param string $size Image size.
+	 */
+	public static function maybe_generate_intermediate_image( $attachment_id, $size ) {
+		$metadata   = image_get_intermediate_size( $attachment_id, $size );
```

Comment:
It seems that `image_get_intermediate_size()` returns a `metadata['path']` like this:

```
2023/10/photo-1680007889201-114ac772447d-Medium-500x500.png
```

Since we are in October, the date in the path is set to 2023/10. I'm wondering if, when we get to November, the path would change to 2023/11. Because if it does change, and we check for the file's existence on line 125, this would result in the images being generated every month. Do you know if this is possible to occur?

Reply:
Good call. I don't know the answer to that right now since we're still in October. I will test it again in a few days to see what happens.

Reply:
If you're running the application locally, I believe that if you change the date in your operating system, you'd be able to test that for different dates. For example, in MacOS:

1. Click the Apple logo > System Preferences.
2. Select Date & Time.
3. Uncheck "Set date and time automatically".
4. Manually adjust the date and time. In this case, you can set the date to November or any other date.

Reply:
I've tested this and the file path remains the same as when the original image size was created.